powerline.el

```diff
@@ -130,9 +130,22 @@ This is needed to make sure that text is properly aligned."
 ;;
 ;; see https://github.com/milkypostman/powerline/issues/58
 ;;
-(defun powerline-delete-cache ()
-  (set-frame-parameter nil 'powerline-cache nil))
-(add-hook 'desktop-save-hook 'powerline-delete-cache)
+;; It is better to put the following code into your init file.
+;; (push '(powerline-cache . :never) frameset-filter-alist)
+;;
+(defun powerline-delete-cache (&optional frame)
+  "Set the FRAME cache to nil."
+  (set-frame-parameter frame 'powerline-cache nil))
+
+(defvar frameset-filter-alist)
```

Comment:
I think it is useful for using `desktop-auto-save` because if `desktop-auto-save-timeout` has integer, caches are deleted when Emacs is idle.

It is unnecessary to clear caches, if `frameset-filter-alist` has `(powerline-cache . :never)`.
If `frameset-filter-alist` has it, `desktop-save`/`desktop-read` don't save/restore `powerline-cache` as a frame parameter because `frameset-save`/`frameset-restore` is called with nil for the argument value of filter and  `frameset-filter-alist` is used as the default filter.

If `frameset-filter-alist` is unnecessary, I'll remove it.